From 4d78a13c714bf48e8717e4559333ca5e6866903c Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 13:19:15 +0530
Subject: [PATCH 024/281] OpenWrt: 310-arm_module_unresolved_weak_sym.patch

fix errors in unresolved weak symbols on arm

lede-commit: 570699d4838a907c3ef9f2819bf19eb72997b32f
Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: I4bee8c81fdfab63f8649186e82ac8849a05136f9
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 arch/arm/kernel/module.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/kernel/module.c b/arch/arm/kernel/module.c
index d59c36dc0494..200a31b0acad 100644
--- a/arch/arm/kernel/module.c
+++ b/arch/arm/kernel/module.c
@@ -146,6 +146,10 @@ apply_relocate(Elf32_Shdr *sechdrs, const char *strtab, unsigned int symindex,
 			return -ENOEXEC;
 		}
 
+		if ((IS_ERR_VALUE(sym->st_value) || !sym->st_value) &&
+		    ELF_ST_BIND(sym->st_info) == STB_WEAK)
+			continue;
+
 		loc = dstsec->sh_addr + rel->r_offset;
 
 		switch (ELF32_R_TYPE(rel->r_info)) {
-- 
2.17.1

